prov/shm/src/smr_init.c

```diff
@@ -63,28 +63,35 @@ static void smr_resolve_addr(const char *node, const char *service,
 	(*addr)[*addrlen - 1]  = '\0';
 }
 
-static int smr_get_ptrace_scope(void)
+static int smr_check_cma_capability(void)
 {
-	FILE *file;
-	int scope, ret;
-
-	scope = 0;
-	file = fopen("/proc/sys/kernel/yama/ptrace_scope", "r");
-	if (file) {
-		ret = fscanf(file, "%d", &scope);
-		if (ret != 1) {
+	pid_t pid;
+	int ret = 0;
+	int flag = 0;
+
+	pid = fork();
+	if (pid == 0) {
+		// child process tries to execute CMA write and exits
+		int cflag = 1;
+		struct iovec local;
+		struct iovec remote;
+
+		local.iov_base = &cflag;
+		local.iov_len = sizeof(int);
+		remote.iov_base = &flag;
+		remote.iov_len = sizeof(int);
+		ret = process_vm_writev(getppid(), &local, 1, &remote, 1, 0);
+		if (ret == -1)
 			FI_WARN(&smr_prov, FI_LOG_CORE,
-				"Error getting value from ptrace_scope\n");
-			return -FI_EINVAL;
-		}
-		ret = fclose(file);
-		if (ret) {
-			FI_WARN(&smr_prov, FI_LOG_CORE,
-				"Error closing ptrace_scope file\n");
-			return -FI_EINVAL;
-		}
+				"Error child trying execute process_vm_writev on its parent: %s\n", strerror(errno));
+		exit(0);
+	} else {
+		// parent process waits child to exit, and check flag bit to see if child's CMA write succeeds or not
```

Comment:
I don't think we need this comment and the comment above.

Reply:
Will remove both comments.

Reply:
My $0.02: comments are good things.  These comments are a _little_ obvious, but they're not harmful.

Reply:
I will bow down to the Squyres God 🙇‍♀ 
Let's keep the comment but at least shorten the line length to keep them under <80 characters per line